homework/index.js

```diff
@@ -29,26 +37,163 @@
+    appendRepoDetail('Updated:', repo.lastUpdateTime, repoDetailsTable);
+  }
+
+  function renderContributorDetails(contributor, ul, index) {
+    const repoContributorListItem = createAndAppend('li', ul, {
+      class: 'contributors-item',
+    });
+    if (index === 0) {
+      createAndAppend('h6', repoContributorListItem, {
+        text: 'Contributions:',
+      });
+    }
+    createAndAppend('img', repoContributorListItem, {
+      src: contributor.avatar_url,
+      alt: 'Contributor avatar picture',
+    });
+    createAndAppend('a', repoContributorListItem, {
+      href: contributor.html_url,
+      text: contributor.login,
+      target: '_blank',
+    });
+    createAndAppend('span', repoContributorListItem, {
+      text: contributor.contributions,
+    });
+  }
+
+  function renderHeaderSection(title, parent) {
+    const header = createAndAppend('header', parent);
+    createAndAppend('span', header, {
+      text: title,
+      class: 'header-title',
+    });
+    return createAndAppend('select', header);
+  }
+
+  function renderSubSection(sectionId, parent) {
+    const subSection = createAndAppend('section', parent, {
+      class: sectionId,
+      id: sectionId,
+    });
+    createAndAppend('ul', subSection, { class: 'sub-container' });
+  }
+
+  function renderMainSectionWithSubSections(parent) {
+    const mainSection = createAndAppend('main', parent, { class: 'container' });
+    renderSubSection(REPOSITORY_DETAIL_SECTION_ID, mainSection);
+    renderSubSection(REPOSITORY_CONTRIBUTORS_SECTION_ID, mainSection);
+  }
+
+  function getUnorderedListAsEmpty(sectionId) {
+    const section = document.getElementById(sectionId);
+    const ul = section.getElementsByTagName('ul')[0];
+    ul.innerHTML = ''; // If the ul has child nodes, clear them first
+    return ul;
+  }
+
+  function renderRepositoryDetailSection(repository) {
+    renderRepoDetails(
+      repository,
+      getUnorderedListAsEmpty(REPOSITORY_DETAIL_SECTION_ID),
+    );
+  }
+
+  function renderRepositoryContributorsSection(repository) {
+    const ul = getUnorderedListAsEmpty(REPOSITORY_CONTRIBUTORS_SECTION_ID);
+    fetchJSON(repository.contributorsUrl).then(contributors => {
+      // In case of non contributors situation
+      if (contributors) {
+        contributors.forEach((contributor, index) => {
+          renderContributorDetails(contributor, ul, index);
+        });
+      }
+    });
+  }
+
+  function renderRepository(repository) {
+    renderRepositoryDetailSection(repository);
+    renderRepositoryContributorsSection(repository);
   }
 
   function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
+    const root = document.getElementById('root');
+    const select = renderHeaderSection('HYF Repositories', root);
+    fetchJSON(url)
+      .then(repos => {
+        const repositoryInfos = sortAndMapRepositories(repos);
```

Comment:
This is just a nit-picking comment, but that there is not plural form for 'information'. See for instance this link:

https://blog.harwardcommunications.com/2010/11/09/information/